forc-plugins/forc-client/src/op/call/transaction_trace.rs

```diff
@@ -0,0 +1,1176 @@
+use crate::op::call::Abi;
+use ansiterm::Color;
+use anyhow::Result;
+use fuel_tx::Receipt;
+use fuels_core::types::ContractId;
+use std::collections::HashMap;
+use std::fmt::{self, Display, Formatter};
+
+/// A node in the transaction trace tree
+pub(crate) struct Node<'a> {
+    receipt: Receipt,
+    children: Vec<Node<'a>>,
+    abis: Option<&'a HashMap<ContractId, Abi>>,
```

Comment:
Without reading the details it feels a little weird that we have an optional hashmap here instead of an empty one.

If the distinction between the empty hashmap and `None` is not important semantically in the rest of the code, maybe we should remove the optional type to simplify the defintion a little bit

Reply:
https://github.com/FuelLabs/sway/pull/7196/commits/d6491b4679eccd24fb01d7b7403c97cc7bfa61e9